packages/grafana-ui/src/components/uPlot/utils.ts

```diff
@@ -51,12 +54,43 @@ export function preparePlotData(frame: DataFrame, ignoreFieldTypes?: FieldType[]
       result.push(f.values.toArray());
       continue;
     }
-
     if (ignoreFieldTypes && ignoreFieldTypes.indexOf(f.type) > -1) {
       continue;
     }
+
+    if (f.config.custom?.stackingMode !== StackingMode.None && f.config.custom?.stackingGroup) {
+      if (!stackingGroups[f.config.custom.stackingGroup]) {
+        stackingGroups[f.config.custom.stackingGroup] = [result.length];
+      } else {
+        stackingGroups[f.config.custom.stackingGroup].push(result.length);
+      }
+    }
```

Comment:
this map is vreated in two places currently - one for results stacking(here) and one for bands creation.